src/dependency-manager/src/manager.ts

```diff
@@ -216,22 +194,217 @@ export default abstract class DependencyManager {
+            node.parameters[key] = (node as any).node_config.parameters[key].default;
+          } else {
+            node.parameters[key] = value;
+          }
+
+          // we copy the new parameter value into the node_config if it doesn't already have it
+          if (node instanceof ServiceNode) {
+            (node.node_config as any).parameters = (node.node_config as any).parameters || {};
+            (node.node_config as any).parameters[key] = (node.node_config as any).parameters[key] || {};
+            (node.node_config as any).parameters[key].default = node.parameters[key];
+          } else if (node instanceof DatastoreNode) {
+            node.node_config.parameters = node.node_config.parameters || {};
+            node.node_config.parameters[key] = node.node_config.parameters[key] || {};
+            node.node_config.parameters[key].default = node.parameters[key];
+          } else if (node instanceof ExternalNode && node.node_config instanceof ServiceConfig) {
+            (node.node_config as any).parameters = (node.node_config as any).parameters || {};
+            (node.node_config as any).parameters[key] = (node.node_config as any).parameters[key] || {};
+            (node.node_config as any).parameters[key].default = node.parameters[key];
+          } else if (node instanceof ExternalNode) {
+            (node.node_config as ServiceDatastore).parameters = (node.node_config as ServiceDatastore).parameters || {};
+            (node.node_config as ServiceDatastore).parameters[key] = (node.node_config as ServiceDatastore).parameters[key] || {};
+            (node.node_config as ServiceDatastore).parameters[key].default = node.parameters[key];
+          }
         }
       }
     }
   }
 
+  private interpolateAllNodeConfigs(graph: DependencyGraph, interface_context: EnvironmentInterfaceContext): void {
+    // map of dependency name (as it is in service config) to normalized_ref
+    // used for lookups in expressions like this: ${ dependencies['friendly/name'].parameters... }
+    const friendly_name_map = ParameterInterpolator.build_friendly_name_map(this.graph);
+
+    let change_detected = true;
+    let passes = 0;
+    const MAX_DEPTH = 100;
+
+    for (const node of this.graph.nodes) {
+      if (node instanceof ServiceNode || (node instanceof ExternalNode && node.node_config instanceof ServiceConfig)) {
+        const serial_config = serialize(node.node_config);
+        const namespaced_serial_config = ParameterInterpolator.namespaceExpressions(node.namespace_ref, serial_config, friendly_name_map[node.ref]);
+        node.node_config = deserialize(ServiceConfigV1, namespaced_serial_config);
+      } else if (node instanceof ExternalNode) {
+        //TODO:76: we can't support interpolation of datastore nodes unless we make ServiceDatastore serializable
```

Comment:
@tjhiggins this is the one open TODO I want to discuss... the `node_config` on the ExternalNode _when it's a ServiceDatastore_ is not serializable. Therefore External Datastores will not support interpolation yet.

Reply:
I see. Semi ignore my other comment. - why is it not serializable?

Reply:
Sorry, meant to say it's not deserializable. It's because ServiceDatastore is an interface.  We could probably change to make it a class unless that has implications that I'm unaware of.

Reply:
Yeah we could make it a class. Still no way to distinguish between ServiceConfig/ServiceDatastore when we transform. We really need to kill datastores...

Reply:
I think I might be okay not supporting ExternalNodes (seems like there are currently issues). I'm going to add a story to kill since its easy to check node.host.

Reply:
https://gitlab.com/architect-io/architect-cli/-/issues/89